src/master/resources/network_bandwidth.cpp

```diff
@@ -0,0 +1,229 @@
+/**
+ * @brief Compute the amount of network bandwidth relative to
+ *   the share of reserved CPU and the network bandwidth declared on the
+ *   slave.
+ *
+ * @param slaveTotalResources The global resources advertised by the slave.
+ * @return The amount of network bandwidth relative to the share of reserved
+ *   CPU.
+ */
+Try<Option<double>> computeNetworkBandwidthBasedOnShareOfCpu(
+  const Resources& taskResources,
+  const Resources& slaveTotalResources) {
+  Option<Resource> totalNetworkBandwidth =
+    getUnreservedResource(slaveTotalResources,
+                          NETWORK_BANDWIDTH_RESOURCE_NAME);
+
+  // No network bandwidth declared in the slave
+  if(totalNetworkBandwidth.isNone()) {
+    LOG(INFO) << "No network bandwidth advertised by the slave.";
+    return Option<double>();
+  }
+
+  Option<Resource> totalCpus =
+    getUnreservedResource(slaveTotalResources, CPUS_RESOURCE_NAME);
+  Option<Resource> reservedCpus =
+    getUnreservedResource(taskResources, CPUS_RESOURCE_NAME);
+
+  if(totalCpus.isNone()) {
+    return Error("No CPU advertised by the slave.");
+  }
+
+  if(reservedCpus.isNone()) {
+    return Error("No CPU declared in the task. " \
+                 "Cannot deduce network bandwidth.");
+  }
+
+  return reservedCpus.get().scalar().value() /
+      totalCpus.get().scalar().value() *
+      totalNetworkBandwidth.get().scalar().value();
+}
+
+
+/**
+ * @brief Get an amount of network bandwidth, if any, from a set of labels.
+ *
+ * @param labels The set of labels to find network bandwidth amount in.
+ * @return The amount of network bandwidth declared in the label if it is
+ *   provided, None if the label is not provided and Error if there was
+ *   an error while extracting the network bandwidth amount.
+ */
+Try<Option<double>> getNetworkBandwidthFromLabel(
+  const Labels& labels) {
+  Option<Label> networkBandwidthLabel =
+    getLabel(labels, NETWORK_BANDWIDTH_LABEL_NAME);
+
+  if(networkBandwidthLabel.isSome()) {
+    LOG(INFO) << "Network bandwidth is specified in a label. " \
+                 "Taking the value.";
+
+    try {
+      // Parse the amount of network bandwidth.
+      return std::stof(
+        networkBandwidthLabel.get().value());
+    }
+    catch(const std::invalid_argument&) {
+      return Error("Invalid network bandwidth resource format. "\
+                   "Should be an integer.");
+    }
+    catch(const std::out_of_range&) {
+      return Error("Network bandwidth amount is out of range.");
+    }
+  }
+  return Option<double>();
+}
+
+
+/**
+ * @brief Enforce network bandwidth allocation (see header for more details).
+ *
+ * @param slaveTotalResources The resources declared on the slave.
+ * @param task The task to enforce network bandwidth for.
+ * @return Nothing if no enforcement is done or if it is successful, otherwise
+ *  an Error.
+ *
+ * TODO(clems4ever): Be able to consume role resources as well as unreserved.
+ */
+Try<Nothing> enforceNetworkBandwidthAllocation(
```

Comment:
maybe the whole thing should be called "default" or "fallback" instead of enforce

Reply:
For me this method does more than defaulting/fallbacking to default value. It ensure a task declares network bandwidth on slaves declaring some, i.e., 
* Does nothing if the task has network bandwidth declared.
* Read the resource from a label if provided.
* Apply default value/Fallback to default value.

So, I think this naming does not really fit with the global purpose of this function.